File: trunk/proj/Android/src/com/jasonzqshen/familyaccounting/core/exception/NoMasterDataFactoryClass.java
package com.jasonzqshen.familyaccounting.core.exception;

import com.jasonzqshen.familyaccounting.core.masterdata.MasterDataType;

public class NoMasterDataFactoryClass extends Exception {
	/**
	 * 
	 */
	private static final long serialVersionUID = 9134563712030765335L;

	public final MasterDataType _type;

	public NoMasterDataFactoryClass(MasterDataType type) {
		super(String.format("Master data type %s is not registered",
				type.toString()));

		_type = type;

	}

}

File: branches/V1.0/proj/WinStore_FamilyAccounting/WinStore_FamilyAccountingCore/Exceptions/NoMasterDataFactoryClass.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WinStore_FamilyAccountingCore.MasterData;

namespace WinStore_FamilyAccountingCore.Exceptions
{
    public class NoMasterDataFactoryClass : Exception
    {
        public readonly MasterDataType MdType;
        public NoMasterDataFactoryClass(MasterDataType type)
            : base(String.Format("Master data type {0} is not registered",
                type.ToString()))
        {
            MdType = type;
        }
    }
}
